### PR TITLE
(closes #1309) fix bug in name comparison in backend

### LINKED ISSUE #1309: [psyir] Function given `result(name)` where `name` is the same as that of the function
This looks like it may simply be a lower/uppercase problem, e.g. we are generating:

    function One_on_L(ptha, pqa, pus, pts, pqs) result(one_on_l)
      real(kind=wp), dimension(jpi,jpj), intent(in) :: ptha
      real(kind=wp), dimension(jpi,jpj), intent(in) :: pqa
      real(kind=wp), dimension(jpi,jpj), intent(in) :: pus
      real(kind=wp), dimension(jpi,jpj), intent(in) :: pts
      real(kind=wp), dimension(jpi,jpj), intent(in) :: pqs
      real(kind=wp), dimension(jpi,jpj) :: one_on_l

from:

    FUNCTION One_on_L( ptha, pqa, pus, pts, pqs )
       REAL(wp), DIMENSION(jpi,jpj)             :: One_on_L         !: 1./(Monin Obukhov length) [m^-1]

### PR DESCRIPTION
**Pull request by arporter**
Tiny bug fix that closes #1309.

### CODE REVIEW

**Comment by arporter** [comment]
2-line change. Ready now :-)

**Comment by arporter** [comment]
Since this is so small, and I suspect the solution to #1306 will be too, I'm taking this back to do #1306.

**Comment by arporter** [comment]
Assuming the CI is happy, this is ready for a look.

File: src/psyclone/tests/psyir/frontend/fparser2_test.py [hunk 1]
```diff
@@ -499,9 +499,10 @@ def test_generate_schedule_dummy_subroutine(parser):
     schedule = processor.generate_schedule("dummy_code", ast)
     assert isinstance(schedule, KernelSchedule)
 
-    # Test argument intent is inferred when not available in the declaration
+    # Test that argument intent is left as UNKNOWN when not available in the
```
Comment (sergisiso) [inline src/psyclone/tests/psyir/frontend/fparser2_test.py:502]: The comment and assert is fine but it seems this check is (and was) misplaced as generate_schedule is not dealing directly with the declarations.

This assert will be better placed in the "test_process_declarations_intent".

**Review by sergisiso — CHANGES_REQUESTED** [review]
@arporter Pycodestyle, pylint and compilation tests and examples are all fine. Just a couple of small comments to address, then this can go to master.

File: src/psyclone/psyir/backend/fortran.py [hunk 2]
```diff
@@ -491,6 +491,12 @@ def gen_vardecl(self, symbol):
         if is_symbol:
             # A member of a derived type cannot have the 'intent' or
             # 'parameter' attribute.
+            # TODO #1306. The fparser2 frontend gives any argument symbols
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/psyir/backend/fortran.py:494]: Deleted!

File: src/psyclone/tests/psyir/frontend/fparser2_test.py (hunk 1 above)
Comment (arporter) [reply src/psyclone/tests/psyir/frontend/fparser2_test.py:502]: Yes, I agree - done.

**Comment by arporter** [comment]
Assuming CI happiness, this is ready for another look now...

**Review by sergisiso — APPROVED** [review]
@arporter Comments addressed, all checks/formatting still passing. This is ready to merge.

**Pull request merged**